qt/applications/workbench/workbench/plotting/figurewindow.py

```diff
@@ -38,6 +40,18 @@ def __init__(self, canvas, parent=None):
         self.setAttribute(Qt.WA_DeleteOnClose, True)
         self.setWindowIcon(QIcon(':/images/MantidIcon.ico'))
 
+        # On Windows, setting the Workbench's main window as this window's
+        # parent always keeps this window on top, but still allows minimization.
+        # On Ubuntu the child is NOT kept above the parent, hence we use the
+        # focusWindowChanged event to bring this window back to the top when
+        # the main window gets focus. This does cause a slight flicker effect
+        # as the window is hidden and quickly brought back to the front. Using
+        # the parent-child method at least avoids this flicker on Windows.
+        if platform.system() == "Windows":
```

Comment:
This is good and does the job but I agree the flicker is not great.

Have you explored the [`Qt::WindowStaysOnTopHint`](https://doc.qt.io/qt-5/qt.html#WindowType-enum) window flag? I've just tried this on Ubuntu with Gnome 3 and also macOS where a simple
```python
self.window.setWindowFlags(self.window.windowFlags() | Qt.WindowStaysOnTopHint)
```
underneath `self.window = FigureWindow(canvas)` in `figuremanager.py` leaves the window on top without any flicker. I wonder if this also works on Windows? It would be much cleaner...

Reply:
I did have a look at that, yeah. However, it keeps the window on top of all windows, not just Workbench. This includes any interfaces and also any other applications you're running.

I tried setting and un-setting the flag when Workbench becomes inactive, but the window needs to be re-drawn whenever new flags are set, which again causes a flicker.

Reply:
Ah I see. I think it might be a good idea to include a comment to this affect as others in the future may wonder the same and it would stop them wasting time trying to do something similar.